packages/xstate-test/src/index.ts

```diff
@@ -306,8 +307,13 @@ export class TestModel<TTestContext, TContext> {
     }
   }
 
-  public getCoverage(): { stateNodes: Record<string, number> } {
-    const stateNodes = getStateNodes(this.machine);
+  public getCoverage(
+    options?: CoverageOptions<TContext>
+  ): { stateNodes: Record<string, number> } {
+    const filter = options ? options.filter : undefined;
+    const stateNodes = getStateNodes(this.machine).filter(
```

Comment:
maybe instead of providing a default filter in a form of `() => true` you could only call filter if needed?
```js
const allStateNodes = getStateNodes(this.machine)
// or rename the variable below to filteredStateNodes, but that would have to be propagated to the code using this
const stateNodes = options.filter ? allStateNodes.filter(options.filter) : allStateNodes
```